pkg/cloudprovider/mesos/mesos.go

```diff
@@ -0,0 +1,238 @@
+	name, err := c.client.clusterName(ctx)
+	return []string{name}, err
+}
+
+// Master gets back the address (either DNS name or IP address) of the leading Mesos master node for the cluster.
+func (c *MesosCloud) Master(clusterName string) (string, error) {
+	clusters, err := c.ListClusters()
+	if err != nil {
+		return "", err
+	}
+	for _, name := range clusters {
+		if name == clusterName {
+			if c.client.master == "" {
+				return "", errors.New("The currently leading master is unknown.")
+			}
+
+			host, _, err := net.SplitHostPort(c.client.master)
+			if err != nil {
+				return "", err
+			}
+
+			return host, nil
+		}
+	}
+	return "", errors.New(fmt.Sprintf("The supplied cluster '%v' does not exist", clusterName))
+}
+
+// ipAddress returns an IP address of the specified instance.
+func ipAddress(name string) (net.IP, error) {
+	if name == "" {
+		return nil, noHostNameSpecified
+	}
+	ipaddr := net.ParseIP(name)
+	if ipaddr != nil {
+		return ipaddr, nil
+	}
+	iplist, err := net.LookupIP(name)
+	if err != nil {
+		log.V(2).Infof("failed to resolve IP from host name '%v': %v", name, err)
+		return nil, err
+	}
+	ipaddr = iplist[0]
+	log.V(2).Infof("resolved host '%v' to '%v'", name, ipaddr)
+	return ipaddr, nil
+}
+
+// ExternalID returns the cloud provider ID of the specified instance.
+func (c *MesosCloud) ExternalID(instance string) (string, error) {
+	ip, err := ipAddress(instance)
+	if err != nil {
+		return "", err
+	}
+	return ip.String(), nil
+}
+
+// List lists instances that match 'filter' which is a regular expression
+// which must match the entire instance name (fqdn).
+func (c *MesosCloud) List(filter string) ([]string, error) {
+	//TODO(jdef) use a timeout here? 15s?
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	nodes, err := c.client.listSlaves(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if len(nodes) == 0 {
+		log.V(2).Info("no slaves found, are any running?")
+		return nil, nil
+	}
+	filterRegex, err := regexp.Compile(filter)
+	if err != nil {
+		return nil, err
+	}
+	addr := []string{}
+	for _, node := range nodes {
+		if filterRegex.MatchString(node.hostname) {
+			addr = append(addr, node.hostname)
+		}
+	}
+	return addr, err
+}
+
+// GetNodeResources gets the resources for a particular node
+func (c *MesosCloud) GetNodeResources(name string) (*api.NodeResources, error) {
+	//TODO(jdef) use a timeout here? 15s?
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	nodes, err := c.client.listSlaves(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if len(nodes) == 0 {
+		log.V(2).Info("no slaves found, are any running?")
+	} else {
+		for _, node := range nodes {
+			if name == node.hostname {
+				return node.resources, nil
+			}
+		}
+	}
+	log.Warningf("failed to locate node spec for %q", name)
+	return nil, nil
+}
+
+// NodeAddresses returns the addresses of the specified instance.
+func (c *MesosCloud) NodeAddresses(name string) ([]api.NodeAddress, error) {
```

Comment:
actually *MesosCloud needs to be the receiver to implement
cloudprovider.Instances

On Thu, May 14, 2015 at 5:12 AM, David Oppenheimer <notifications@github.com
> wrote:

> In pkg/cloudprovider/mesos/mesos.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8053#discussion_r30307422>
> :
>
> > +	}
> > +	if len(nodes) == 0 {
> > +		log.V(2).Info("no slaves found, are any running?")
> > +	} else {
> > +		for _, node := range nodes {
> > +			if name == node.hostname {
> > +				return node.resources, nil
> > +			}
> > +		}
> > +	}
> > +	log.Warningf("failed to locate node spec for %q", name)
> > +	return nil, nil
> > +}
> > +
> > +// NodeAddresses returns the addresses of the specified instance.
> > +func (c *MesosCloud) NodeAddresses(name string) ([]api.NodeAddress, error) {
>
> same here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8053/files#r30307422>
> .
>